services/apps/alcs/src/alcs/inquiry/inquiry.service.spec.ts

```diff
@@ -0,0 +1,249 @@
+import { DeepMocked, createMock } from '@golevelup/nestjs-testing';
+import { Test, TestingModule } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { classes } from 'automapper-classes';
+import { AutomapperModule } from 'automapper-nestjs';
+import { Repository } from 'typeorm';
+import { ServiceNotFoundException } from '../../../../../libs/common/src/exceptions/base.exception';
+import { InquiryProfile } from '../../common/automapper/inquiry.automapper.profile';
+import { FileNumberService } from '../../file-number/file-number.service';
+import { Board } from '../board/board.entity';
+import { Card } from '../card/card.entity';
+import { CardService } from '../card/card.service';
+import { InquiryParcel } from './inquiry-parcel/inquiry-parcel.entity';
+import { InquiryType } from './inquiry-type.entity';
+import { UpdateInquiryDto } from './inquiry.dto';
+import { Inquiry } from './inquiry.entity';
+import { InquiryService } from './inquiry.service';
+
+describe('InquiryService', () => {
+  let service: InquiryService;
+  let mockCardService: DeepMocked<CardService>;
+  let mockRepository: DeepMocked<Repository<Inquiry>>;
+  let mockTypeRepository: DeepMocked<Repository<InquiryType>>;
+  let mockFileNumberService: DeepMocked<FileNumberService>;
+
+  beforeEach(async () => {
+    mockCardService = createMock();
+    mockRepository = createMock();
+    mockTypeRepository = createMock();
+    mockFileNumberService = createMock();
+
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        AutomapperModule.forRoot({
+          strategyInitializer: classes(),
+        }),
+      ],
+      providers: [
+        InquiryService,
+        InquiryProfile,
+        {
+          provide: CardService,
+          useValue: mockCardService,
+        },
+        {
+          provide: getRepositoryToken(Inquiry),
+          useValue: mockRepository,
+        },
+        {
+          provide: getRepositoryToken(InquiryType),
+          useValue: mockTypeRepository,
+        },
+        {
+          provide: FileNumberService,
+          useValue: mockFileNumberService,
+        },
+      ],
+    }).compile();
+
+    service = module.get<InquiryService>(InquiryService);
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  it('should load the type code and call the repo to save when creating', async () => {
+    const mockCard = {} as Card;
```

Comment:
nitpick: new Card() and new Board() instead of casting

Reply:
done